### ISSUE TITLE
as_sql() got an unexpected keyword argument 'subquery'

### ISSUE THREAD

**Issue by benmehlman** [lionheart/django-pyodbc#143]
Using django==1.10.6, django-pyodbc==1.1.0, pyodbc==4.0.16

Django has added an argument `subquery=False` to `SQLCompiler.as_sql()`.  This argument is not present in the corresponding django-pyodbc `SQLCompiler.as_sql()` subclass implementation.  This causes a TypeError when using subqueries.

Patch:

~~~
--- /opt/env/ssbc/local/lib/python2.7/site-packages/django_pyodbc/compiler.py.orig      2017-04-07 23:20:22.860313700 -0400
+++ /opt/env/ssbc/local/lib/python2.7/site-packages/django_pyodbc/compiler.py   2017-04-07 23:09:21.143932644 -0400
@@ -196,7 +196,7 @@
             elif aggregate.sql_function == 'VAR_POP':
                 self.query.aggregate_select[alias].sql_function = 'VARP'

-    def as_sql(self, with_limits=True, with_col_aliases=False):
+    def as_sql(self, with_limits=True, with_col_aliases=False, subquery=False):
         # Django #12192 - Don't execute any DB query when QS slicing results in limit 0
         if with_limits and self.query.low_mark == self.query.high_mark:
             return '', ()
@@ -213,7 +213,7 @@
             # unless TOP or FOR XML is also specified.
             try:
                 setattr(self.query, '_mssql_ordering_not_allowed', with_col_aliases)
-                result = super(SQLCompiler, self).as_sql(with_limits, with_col_aliases)
+                result = super(SQLCompiler, self).as_sql(with_limits, with_col_aliases, subquery)
             finally:
                 # remove in case query is every reused
                 delattr(self.query, '_mssql_ordering_not_allowed')
~~~

Thank you!
Ben Mehlman

**Comment by dlo** [comment]
Hi Ben! Thanks for the report. I'd be happy to merge a PR. Have you tested this change with older versions of Django? Thanks!

**Comment by benmehlman** [comment]
I haven't checked with other versions but can easily do a 'git blame' to find out what version of Django added that argument.  In our environment we are doing a huge leap from django 1.6 to 1.10 hence discovering this problem.

Is there a minimum version of django you want to keep django-pyodbc compatible with?  I could recode this passing **kwargs instead of an  explicit argument so that it would have more backward compatibility...

Thanks!

**Comment by benmehlman** [comment]
Hi Dan, having looked a little more into this now it unfortunately not that simple.. this particular method was changed in django 1.8, then changed BACK in django 1.11.. so django-pyodbc is broken with django 1.8, 1.9, and 1.10 but now isn't broken anymore due to them undoing the change in django.

So although this particular problem disappears in 1.11, if it's desired for django-pyodbc to retain compatibility with 1.8-1.10 then it would be necessary to change this patch to use **kwargs so that it will pass the extra arguments if they are present, and if not, do nothing.

If that's ok with you.

**Comment by dlo** [comment]
Merged in #145, thanks @benmehlman for the patch!
